okhttp/src/main/java/io/grpc/okhttp/OkHttpChannelBuilder.java

```diff
@@ -252,12 +257,33 @@ SSLSocketFactory createSocketFactory() {
       case TLS:
         try {
           if (sslSocketFactory == null) {
-            SSLContext sslContext = SSLContext.getInstance("Default", Platform.get().getProvider());
+            SSLContext sslContext = null;
+            if (GrpcUtil.IS_RESTRICTED_APPENGINE) {
+              // The following auth code circumvents the following AccessControlException:
```

Comment:
This comment is good, but it doesn't really explain why or how.  Where is that exception thrown from?  Why does this block of code change that?